src/buttons/Button.js

```diff
@@ -53,6 +52,12 @@ class Button extends Component {
       ...attributes
     } = this.props;
 
+    let { textStyle = {} } = this.props;
+    textStyle =
+      buttonStyle && buttonStyle.width && !textStyle.width
+        ? Object.assign({}, textStyle, { width: '100%' })
```

Comment:
You could use directly this
```jsx
const textStyle = {
  buttonStyle && buttonStyle.width && { width: '100%' },
  ...textStyle,
}
```
If there is a width in `textStyle`, it will override it automatically

Reply:
@xavier-villelegier 
`!textStyle.width`
this check right here makes sure we don't do the override when a width is specified on textStyle ?

Reply:
Yea sure, your code works fine ! I was just proposing a less verbose more compact solution, using the `...` syntax and the override functionnality with js object 👍

Here I don't need to check for `textStyle.width`, as the `textStyle` object will override the `width: '100%'` automatically if it contains any `width` in it.

Reply:
@xavier-villelegier sure I will do that asap !!!!! 
```javascript
const textStyle = {
  buttonStyle && buttonStyle.width && { width: '100%' },
  ...textStyle,
}
```
- should we not spread `...{width: 100%}` as well
- Also `buttonStyle && buttonStyle.width && { width: '100%' }` modelling this as 
   `true && true && {width: '100%'}` throws an error in the console
   `Uncaught SyntaxError: Unexpected token && `
     dirty work around would be to use the computed object property syntax

Reply:
@Gregjarvez Ah yea sorry I meant
```jsx
const textStyle = {
  width: buttonStyle && buttonStyle.width ? '100%' : null,
  ...textStyle,
}
```

My bad 😅

Reply:
lol marvellous I did that already !! I have also pulled the changes from next .